board/ipaccess/ipa9131/test_board.c

```diff
@@ -0,0 +1,84 @@
+#include <common.h>
+#include <linux/compiler.h>
+#include <i2c.h>
+
+#if defined CONFIG_CMD_IPA9131_TEST_BOARD
+
+static int is_test_board(void)
+{
+    uint8_t buf = 0;
+    if ( 0 != i2c_read(CONFIG_CHARACTERISATION_EEPROM_ADDR,
+                CONFIG_IPA9131_MISC_FLAGS_OFFSET,
+                1,
+                &buf,
+                1) )
+    {
+        fprintf(stderr,"Error Reading test board bit\n");
+        return -1;
+    }
+    return (int) (buf & 0x01);
+}
+
+static int toggle_test_board(void)
+{
+    uint8_t buf = 0;
+    int ret = 0;
+    if ( -1 != (ret = is_test_board()) )
+    {
+        buf = !ret;
+    }
+    else
+    {
+        fprintf(stderr,"Error checking current status of board\n");
+        return -1;
+    }
+    if ( 0 != i2c_write(CONFIG_CHARACTERISATION_EEPROM_ADDR,
+                CONFIG_IPA9131_MISC_FLAGS_OFFSET,
+                1,
+                &buf,
+                1) )
+    {
+        fprintf(stderr,"Error writing test board bit\n");
+        return -1;
+    }
+    return 0;
+}
+
+int do_test_board(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
```

Comment:
This does not feel right.

This can only be fiddled when the board is set up as a dev board (in fuses) - it must just fail otherwise.

Also, "toggle" is not really usable, as there's no "query" beforehand - I'd prefer to just see "set" and "clear".